back/engines/commercial/insights/app/controllers/insights/web_api/v1/classification_tasks_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module Insights
+  module WebApi
+    module V1
+      class ClassificationTasksController < ::ApplicationController
+        skip_after_action :verify_policy_scoped, only: [:index]
+        after_action :verify_authorized, only: [:index]
+
+        def index
+          tasks = ZeroshotClassificationTasksFinder.new(
+            categories || view.categories, # use all the categories if the query parameter is not provided
+            inputs: inputs
+          ).execute
+          
+          render json: ZeroshotClassificationTaskSerializer.new(tasks, params: fastjson_params), status: :ok
+        end
+
+        def create
+          Insights::CreateClassificationTasksJob.perform_later(
```

Comment:
Perform later ? On category creation or in scan button I think we'll want to trigger this immediately, and that's the only places I can think of that could use this, no ? Or is this to be used by nlp ?

Reply:
Yeah, I wanted to make to make it snappy! (Even if the endpoint is for zeroshot classification is async, we still need to go through all the inputs in the scope and send them in a big request.) In the end, it's not working very well because the FE expects the tasks to be there. So I will rework it.

What is the scan button?

Reply:
When you create a category as a premium customer, you get a scan button that uses zero-shot to provide you with inputs for that category (=which assigns suggestions to inputs that match that category)

![image](https://user-images.githubusercontent.com/38424075/125769145-d851a24c-86de-499a-a2fc-f5776c8755c8.png)

Reply:
(An I think we should have this button even when it's not empty, under the tree dots, but I don't find back info about that in the specs)

Reply:
Ah yeah, that one. Forgot it's called "scan".